MDEV-23538: Rename mariadb.pc to mariadb-embedded-server.pc to avoid confusion

ottok opened the pull request:
https://jira.mariadb.org/browse/MDEV-23538

Change so that the client library config can be accessed using the name
libmariadb.pc or mariadb.pc. Rename old mariadb.pc to mariadbd.pc to better
indicate that the config is for the daemon build, not client.

In downstream Debian we already ship the libmariadb.pc as mariadb.pc:
https://salsa.debian.org/mariadb-team/mariadb-10.5/-/commit/2f183af990fbe1cfa8c343998c7640f45f45368b

cvicentiu commented:
Hi @ottok !

Is this still WIP, or ready for review?

ottok commented:
Thanks for taking a look! Serg suggested removing the libmariadb symlink in https://jira.mariadb.org/browse/MDEV-23538. The PR is technically done, we just need to get a technical decision on the naming, see Jira.

My suggestion is this PR.

ottok commented:
And yeah, the `debian/rules` would be better done in CMake. I can close this PR if you want to take a shot at putting the files with correct names in correct places directly in CMake code.

illuusio commented:
> And yeah, the `debian/rules` would be better done in CMake. I can close this PR if you want to take a shot at putting the files with correct names in correct places directly in CMake code.

This should be done in CMakeFiles so it would be benefitting RPM also.

ottok commented:
@illuusio If you make a CMake level implementation and post the PR url here, I will close this PR in favor of yours.

ottok commented:
> @illuusio If you make a CMake level implementation and post the PR url here, I will close this PR in favor of yours.

Ideally this would be fixed on the CMake level. Never the less, I also now rebased this on latest 10.9.

Also, due to @vuvova comment in https://jira.mariadb.org/browse/MDEV-23538 I changed the new name from `mariadbd.pc` to `libmariadbd.pc`.

ottok commented:
Rebased on 11.0 now. It was already fine to merge in 10.9, hopefully it will get merged now.

illuusio commented on debian/libmariadb-dev.install:
```diff
@@ -12,10 +12,10 @@ usr/include/mariadb/mariadb_dyncol.h
 usr/include/mariadb/my_config.h
 usr/include/mariadb/my_global.h
 usr/include/mariadb/my_sys.h
-usr/include/mariadb/my_alloca.h
```
This does not belong here?

ottok replied:
It was already here, I just ran `wrap-and-sort -v` to have the file listed alphabetically.

illuusio commented:
There is problem that most of the code does not compile on Debian withouth patching and compiles on other distros fine or am I missing something?

ottok commented:
> There is problem that most of the code does not compile on Debian withouth patching and compiles on other distros fine or am I missing something?

CI currently failing on test `innodb_zip.cmp_drop_table` and build error on `columnstore.dir/ha_mcs_dml.cpp.o] Error 1`. I do not see how they can be caused by the changes in this PR - seems more likely that the CI on branch 11.0 is unstable.

vuvova reviewed (changes requested):
@ottok, I don't understand. You rename `support-files/mariadb.pc` to be installed as `libmariadbd.pc` ?
But its content is
```
$ grep Libs support-files/mariadb.pc
Libs: -L${libdir} -lmariadb -ldl -lm -lssl -lcrypto -lz
```
it's for a client library, not for embedded server. It might make sense to remove it completely. But to install it as `libmariadbd.pc` looks totally wrong to me.

ottok commented:
@vuvova There is https://github.com/mariadb-corporation/mariadb-connector-c/blob/3.3/mariadb_config/libmariadb.pc.in for the client, and this file is being built and shipped as libmariadb.pc. Then we have this server file you referenced that comes from https://github.com/MariaDB/server/blob/11.1/support-files/mariadb.pc.in. MariaDB should not have two `*mariadb.pc'. The other one should be '*mariadbd.pc` or if you think nobody ever used it for building the embedded server then the solution would be just to delete the file as a duplicate?

LinuxJedi reviewed (changes requested):
I did a quick scan of other pkgconfig files installed in distros and came to a similar conclusion to Sergei in the Jira ticket.

If this is to be renamed I recommend using `mariadbd.pc` or something more verbose such as `mariadb-embedded-server.pc`. Very likely (at least initially) a symlink would be needed too, so that compatibility isn't broken. This is probably something to be discussed in the Jira ticket though.

ottok commented:
Rebased on latest 11.2 and also changed to use name `mariadb-embedded-server.pc'.

LinuxJedi commented:
I think this looks OK now, but I defer to @vuvova's opinion here.

ottok commented:
Rebased on latest 11.3 but CI seems to fail on tests `main.analyze_stmt_slow_query_log` and `perfschema.events_waits_current_MDEV-29091`. I have not checked if these tests were already broken before the changes. Most of the commits on https://github.com/MariaDB/server/commits/11.3 are currently not passing CI (I see a lot of red crosses attached to them).

vuvova reviewed (approved):
To my inexperienced eye it looks fine. I'm slightly concerned that out of 671 `*.pc` files on my laptop, not a single one is a symlink to a name with (or without) `lib` prefix, so this is **really** unconventional.

Still, there are three symlinks there:
```
$ find  /usr/share/pkgconfig/ /usr/lib/pkgconfig/ /usr/lib64/pkgconfig/ -type l -ls
8673227  0 lrwxrwxrwx  1 root  root  12 May  4 16:49 /usr/lib64/pkgconfig/libcrypt.pc -> libxcrypt.pc
8671475  0 lrwxrwxrwx  1 root  root  11 Sep  2 15:19 /usr/lib64/pkgconfig/babl.pc -> babl-0.1.pc
8791259  0 lrwxrwxrwx  1 root  root  11 Sep  2 15:00 /usr/lib64/pkgconfig/libpng.pc -> libpng16.pc
```
so at least it's possible and `pkg-config` won't crash on that.

ottok commented:
Rebased on latest 11.3.

LinuxJedi commented on debian/rules:
```diff
@@ -175,6 +175,17 @@ endif
 
+	# Rename mariadb.pc as mariadb-embedded-server.pc as to be more descriptive
+	mv -v $(TMP)/usr/lib/$(DEB_HOST_MULTIARCH)/pkgconfig/mariadb.pc $(TMP)/usr/lib/$(DEB_HOST_MULTIARCH)/pkgconfig/mariadb-embedded-server.pc
+	# Keep libmariabdd.pc around for backwards compatibility as all Debian users
```
Minor typo here.

ottok replied:
Fixed `libmariabdd`

LinuxJedi reviewed (commented):
@ottok I don't appear to have permissions to rebase this for some reason, and there is a minor typo, if you could fix it, we should be good to go.

ottok commented:
Downstream https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1024241 is also related.

I rebased this on 'main' and also updated the target branch to 'main' now. In the future this won't need an update every couple of months.
